src/duckling/migration/migration.service.ts

```diff
@@ -72,16 +73,49 @@ export class MigrationService {
             throw new Error(`You should update duckling. The project expects editor version ${versionFile.editorVersion}. The current version is  ${EDITOR_VERSION}`);
         }
 
+        await this._addMissingEditorMigrations(versionFile, this._findMissingEditorMigrations(versionFile.mapMigrations), versionFileName);
+
         return {
             mapMigrations : versionFile.mapMigrations,
             mapVersion : versionFile.projectVersion
         }
     }
 
+    private async _addMissingEditorMigrations(versionFile : VersionFile, missingMigrations : MapMigration[], versionFileName : string) {
+        versionFile.mapMigrations = versionFile.mapMigrations.concat(missingMigrations);
+        for (let migration of missingMigrations) {
+            versionFile.projectVersion = incrementMajorVersion(versionFile.projectVersion);
+            migration.updateTo = versionFile.projectVersion;
+        }
+        versionFile.editorVersion = EDITOR_VERSION;
+        await this._jsonLoader.saveJsonToPath(versionFileName, JSON.stringify(versionFile, null, 4));
+    }
+
+    private _findMissingEditorMigrations(currentMapMigrations : MapMigration[]) : MapMigration[] {
+        let missingMigrations : MapMigration[] = [];
+        for (let editorMigration of editorMigrations) {
+            if (!this._isMigrationPresent(editorMigration.name, currentMapMigrations)) {
+                missingMigrations.push(editorMigration);
+            }
+        }
+        return missingMigrations;
```

Comment:
return currentMapMigrations.filter(migration => !this._migrationPresent(editorMigration.name, currentMapMigrations));